mentat/parsers/file_edit.py

```diff
@@ -0,0 +1,169 @@
+from __future__ import annotations
+
+import os
+from pathlib import Path
+from typing import TYPE_CHECKING
+
+import attr
+from termcolor import cprint
+
+from mentat.config_manager import ConfigManager
+from mentat.errors import MentatError
+from mentat.parsers.change_display_helper import (
+    DisplayInformation,
+    FileActionType,
+    get_full_change,
+)
+from mentat.user_input_manager import UserInputManager
+
+if TYPE_CHECKING:
+    # This normally will cause a circular import
+    from mentat.code_file_manager import CodeFileManager
```

Comment:
I agree, it would be good to clean up our code so it isn't so interconnected; we might be able to get it done after refactoring the interface and our singletons.